src/api/posts.ts

```diff
@@ -0,0 +1,6 @@
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+export const getPosts = (userId: number) => {
+  return client.get<Post[]>(`/posts?userId=${userId}`);
```

Comment:
Consider adding error handling for the `getPosts` API call. Adding a `.catch` block will help manage potential errors during the request and provide feedback to the user if the post retrieval fails.